src/v/cloud_storage/remote_partition.cc

```diff
@@ -484,7 +490,39 @@ remote_partition::remote_partition(
   , _bucket(std::move(bucket))
   , _probe(m.get_ntp()) {}
 
-ss::future<> remote_partition::start() { co_return; }
+ss::future<> remote_partition::start() {
+    // Fiber that consumers from _eviction_list and calls stop on items before
+    // destroying them.
+    ssx::spawn_with_gate(_gate, [this] { return run_eviction_loop(); });
+    co_return;
+}
+
+void remote_partition::evict_reader(
+  std::unique_ptr<remote_segment_batch_reader> reader) {
+    _eviction_pending.push_back(std::move(reader));
+    _has_evictions_cvar.signal();
+}
+
+void remote_partition::evict_segment(
+  ss::lw_shared_ptr<remote_segment> segment) {
+    _eviction_pending.push_back(std::move(segment));
+    _has_evictions_cvar.signal();
+}
+
+ss::future<> remote_partition::run_eviction_loop() {
+    // Evict readers asynchronously.
+    // NOTE: exits when the condition variable is broken.
+    while (true) {
+        co_await _has_evictions_cvar.wait(
+          [this] { return !_eviction_pending.empty(); });
+        auto eviction_in_flight = std::exchange(_eviction_pending, {});
+        co_await ss::max_concurrent_for_each(
+          eviction_in_flight, 200, [](auto&& rs_variant) {
```

Comment:
Nit: this concurrency is reduced because the eviction has moved to per partition, so should this limit be dynamically calculated to adjust for the number of remote partitions?

Reply:
It's an interesting question. If there were a target number of concurrent deletions, I think it'd be an easier call to make, but even the original 1024 total concurrent evictions per shard was a bit magical.

One saving grace that helps limit the per-shard number of evictions is that the number of partitions itself is limited, as is the total number of materialized segments.

Will leave this as is unless you feel strongly about it.